### PR TITLE
IC

### PR DESCRIPTION
**Pull request by MikeLippincott**
This PR runs IC on the image data

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        args: [--exclude, black, --keep, isort]
+  # Formats import order
+-   repo: https://github.com/pycqa/isort
+    rev: 5.12.0
+    hooks:
+    -   id: isort
+        name: isort (python)
+        args: ["--profile", "black", "--filter-files"]
+
+  #Code formatter for both python files and jupyter notebooks
```
Comment (jenna-tomkinson) [inline .pre-commit-config.yaml]:
```suggestion
  # Code formatter for both python files and jupyter notebooks
```

File: Wave1_data/2.illumination_correction/.DS_Store
Comment (jenna-tomkinson) [inline Wave1_data/2.illumination_correction/.DS_Store:1]: Recommend gitignoring these from Mac.

File: Wave1_data/2.illumination_correction/scripts/0.perform_ic.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Run CellProfiler `illum.cppipe` (IC) pipeline
+#
+# In this notebook, we run the CellProfiler IC pipeline to calculate the illumination (illum) correction functions for all images per channel (5), apply the functions, and save images into a new directory.
```
Comment (jenna-tomkinson) [inline Wave1_data/2.illumination_correction/scripts/0.perform_ic.py:6]: Just to confirm, 5 channels is correct? Recommend including what the channels are in this notebook or in a README.

File: Wave1_data/2.illumination_correction/scripts/0.perform_ic.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+# # get 5 random directories with images
+# import random
+
+# import numpy as np
+
+# random.seed(0)
+# DMSO_sample = raw_directories[5:20]
+# raw_directories = random.sample(raw_directories, 2)
+# raw_directories = DMSO_sample + raw_directories
+# #####################################
```
Comment (jenna-tomkinson) [inline Wave1_data/2.illumination_correction/scripts/0.perform_ic.py]: Do you need this commented-out code anymore or will this be removed in another PR?

File: Wave1_data/2.illumination_correction/pipelines/illum_5ch.cppipe [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+    Regular expression to extract from folder name:(?P<Date>[0-9]{4}_[0-9]{2}_[0-9]{2})$
+    Extract metadata from:All images
+    Select the filtering criteria:and (file does contain "")
+    Metadata file location:Elsewhere...|
+    Match file and image metadata:[]
+    Use case insensitive matching?:No
+    Metadata file name:None
+    Does cached metadata exist?:No
+
+NamesAndTypes:[module_num:3|svn_version:'Unknown'|variable_revision_number:8|show_window:False|notes:['This pipeline is meant for correction with 4 channels specifically:', '', 'DAPI', 'GFP', 'RFP', 'CY5', '']|batch_state:array([], dtype=uint8)|enabled:True|wants_pause:False]
```
Comment (jenna-tomkinson) [inline Wave1_data/2.illumination_correction/pipelines/illum_5ch.cppipe]: Recommend updating this notes in the GUI to avoid any confusion since this info is related to the NF1 project. Updating this to the correct info like metadata channel 1 = DNA, etc.

File: Wave1_data/2.illumination_correction/pipelines/illum_5ch.cppipe [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+    Select the image type:Grayscale image
+    Set intensity range from:Image metadata
+    Maximum intensity:255.0
+
+Groups:[module_num:4|svn_version:'Unknown'|variable_revision_number:2|show_window:False|notes:['']|batch_state:array([], dtype=uint8)|enabled:True|wants_pause:False]
+    Do you want to group your images?:Yes
+    grouping metadata count:1
+    Metadata category:Well
+
+MeasureImageQuality:[module_num:5|svn_version:'Unknown'|variable_revision_number:6|show_window:False|notes:['Only calculate the blur and saturation image quality metrics as we are using these to flag images for poor quality. Blur is set as default.']|batch_state:array([], dtype=uint8)|enabled:True|wants_pause:False]
```
Comment (jenna-tomkinson) [inline Wave1_data/2.illumination_correction/pipelines/illum_5ch.cppipe]: I recommend removing this module since you are not using FlagImages or are looking to perform any whole image QC before IC. It will take up compute time you don't end up needing/using.

File: Wave1_data/2.illumination_correction/pipelines/illum_5ch.cppipe [hunk 6]
```diff
@@ -0,0 +1,370 @@
...
+    Calculate intensity metrics?:No
+    Calculate thresholds?:No
+    Use all thresholding methods?:No
+    Select a thresholding method:Otsu
+    Typical fraction of the image covered by objects:0.1
+    Two-class or three-class thresholding?:Two classes
+    Minimize the weighted variance or the entropy?:Weighted variance
+    Assign pixels in the middle intensity class to the foreground or the background?:Foreground
+
+CorrectIlluminationCalculate:[module_num:6|svn_version:'Unknown'|variable_revision_number:2|show_window:False|notes:['Since for all seeding densities, Schwann cells do not take up a ton of space in the images, we use Background in the IC calculations.', '', 'Block size 50 seems to work best for the DAPI channel (nuclei), as well as using the Fit Polynomial smoothing method as it seemed to not over correct as compared to the other methods.', '', 'We use the each method for calculating IC functions as this seemed to produce the best results over creating one from all images in a channel.']|batch_state:array([], dtype=uint8)|enabled:True|wants_pause:False]
```
Comment (jenna-tomkinson) [inline Wave1_data/2.illumination_correction/pipelines/illum_5ch.cppipe]: For all of the CorrectIlluminationCalculate, I recommend removing the notes from each of these since they don't provide any relevant context to this pyroptosis data. For example, it references Schwann cells which are not the cell line in this project. It just avoids confusion since someone wanting to reproduce this will see these notes in the GUI.

**Review by jenna-tomkinson — APPROVED** [review]
Really nice short and concise PR! I have added some comments for you to address, and I highly recommend adding a README for this module before you merge where you can add the Alpine info and how long IC took on Alpine (which I am very curious about!).  Exciting work!

File: Wave1_data/2.illumination_correction/.DS_Store
Comment (MikeLippincott) [reply Wave1_data/2.illumination_correction/.DS_Store:1]: gitignored

File: Wave1_data/2.illumination_correction/scripts/0.perform_ic.py (hunk 2 above)
Comment (MikeLippincott) [reply Wave1_data/2.illumination_correction/scripts/0.perform_ic.py:6]: Added more readme documentation

File: Wave1_data/2.illumination_correction/scripts/0.perform_ic.py (hunk 3 above)
Comment (MikeLippincott) [reply Wave1_data/2.illumination_correction/scripts/0.perform_ic.py]: Removed :D

File: Wave1_data/2.illumination_correction/pipelines/illum_5ch.cppipe (hunk 5 above)
Comment (MikeLippincott) [reply Wave1_data/2.illumination_correction/pipelines/illum_5ch.cppipe]: Removed!

File: Wave1_data/2.illumination_correction/pipelines/illum_5ch.cppipe (hunk 6 above)
Comment (MikeLippincott) [reply Wave1_data/2.illumination_correction/pipelines/illum_5ch.cppipe]: Removed!

File: Wave1_data/2.illumination_correction/pipelines/illum_5ch.cppipe (hunk 4 above)
Comment (MikeLippincott) [reply Wave1_data/2.illumination_correction/pipelines/illum_5ch.cppipe]: renamed!
